### ISSUE TITLE
Proposed dev priorities

### ISSUE THREAD

**Issue by ManfredKarrer** [bisq-network/proposals#91]
> _This is a Bisq Network proposal. Please familiarize yourself with the [submission and review process](https://docs.bisq.network/proposals.html)._

Here are the dev priorities I would propose for the next few months:

1. Implement protection tools (#78, #83) and once that is in place remove the 0.01 BTC limitation for new accounts buying BTC. 

2. Add P2P chat for traders (#90)

3. New trade protocol

4. API (best to be done in parallet to the above)

5. Monero tx proof verification (#86)

6. Self certification of bank account details using ID verified digital certificates(#79)

I also would suggest to start to offer bounties but those require first a solid tech spec to defined the requirements.

EDIT by @mpolavieja:   API and BitcoinJ 0.15 will be done in paralell with all other priorities.

**Comment by blabno** [comment]
I vote for the API cause it's almost there, just needs to align with changes in core. And the longer we keep it in WIP mode the more effort is wasted on re-syncing with master.

**Comment by ManfredKarrer** [comment]
@blabno Yes agree. I just cannot be available anymore for doing the review and testing. There have been some new features and important changes in the last 2 months, so you need to get those added probably. Hope @sqrrm @devinbileck or @ripcurlx can help with review/testing.

**Comment by ManfredKarrer** [comment]
@mpolavieja Would you like to kick start a dev session next tuesday to get things rolling? Would be great if most devs can join to see how things get organized....

**Comment by devinbileck** [comment]
Sure, I will do my best to help out with the API.

**Comment by mpolavieja** [comment]
I don´t have enough knowledge about the project and how the development works, I´d rather prefer to first look, listen and learn from dev sessions while I try to write the tech spec for protection tools.  If the developers here have examples or templates **they like** for tech specs that would be great to have.

**Comment by sqrrm** [comment]
@blabno I think it would be good to get the API live as soon as possible. It's been a long time coming and as you say, it's more wasted effort the longer it's kept in parallel unmerged. I was hoping Manfred would get it merged before leaving but as that didn't happen I'll try to help. Best if you could recap where it's at to guide us into it.

**Comment by devinbileck** [comment]
Perhaps we should also try to prioritize BitcoinJ 0.15 as well so it doesn't drag on either, since it appears to be ready. Just needs lots of testing - I can focus on that.

**Comment by HarryMacfinned** [comment]
I (am sorry to) disagree about having protection tools at the top of priorities.
As I already exposed it, imo this is running after elusive users.
The most we try to protect them (largely from themselves and the fiat banking system) the more we push them on irresponsability and need for (unscalable) support.
Not speaking about the fact that having some Bisq mechanisms entering this area has probably some dangerous legal issues.
Not speaking also about the fact that trying to correct from outside a system which is **crappy by design** is a never ending story.

I know (and I'm happy) that Bisq first goal was to put *"fiat on ramp"*.
But, after 3.5 years of operation, what is seeable from the numbers is that fiat on Bisq is in majority, a great number but of tiny trades (+ decreasing volume from 2017->2018).
It's more a game than serious **useful** volume (reported to the number of users).
(This has imo several deep reasons which have no reason to change nextly.)

We see those weeks that BSQ, which is a **simple token** and not even a coin, which has itself small volume ... is nevertheless already challenging EUR and USD !! which are amid the most used currencies in the world. This says very long about the true sizes of the concerned communities.

I bet that the next protection tools will not increase the fiat trade volume, but how nice the tools will be, this will help the fiat volume to still decrease. The more we try to protect fiat traders, the more those protections (because of the technicity/delays/limits) make them run away.

When I discuss about Bisq with some friends who trade, they always tell me that Bisq is already too complicated. And some of those guys are technical and should really imo be comfortable with Bisq.
I don't see how adding steps in the current process can arrange things.

Imo the best protection for users should be to tell them that if they lose money due to chargebacks, that's simply outside the scope of Bisq (which is the reality), and the loss is for them. Period.

On its nice path, Bisq got the opportunity (which has nothing to do with hazard) to cross the road and needs of some nice cryptocoins.
The users of those cryptocoins are also real life users. They merit imo also their fair part of care, ie more care than fiat/fun users.
It's those guys who are possibly building a world with functional cryptos and thus it's absolutely unavoidable that those users are also the future of Bisq (if Bisq dies not before).
There is few chances that peoples still holding to back and forth playing fiat/crypto represent the future.

Fiat world and crypto world are distinct (and **diverging**) worlds, because of the **willingness** of the fiat world. Users who still want to have their feet on both world use centralized exchange at 99.x%.
And, even if we see regularly centralized exchanges being stripped ... guess what, users **continue** stubornly to use those platform.
N times we have seen users being scammed on large scales. At each time one of us wrote *"ah, Bisq will get new users"* ... but in fact no. This really simply doesn't happen.

From previous professional experiences, there is no entity able to survive very long with its ressources inversely affected to its customer segments. 90% volume getting 10% of dev, and 10% volume getting 90% of dev is not sustainable in the long run.

Sorry for the WoT, but having some experience about those kind of trials, I cannot stay silent.

**Comment by ManfredKarrer** [comment]
>Perhaps we should also try to prioritize BitcoinJ 0.15 as well so it doesn't drag on either, since it appears to be ready. Just needs lots of testing - I can focus on that.

I reviewed it and looked all good to me, but would be good if someone else have a review as well (bitcoinJ repo) and yes it requires very good testing, specially also the DAO part.

**Comment by sqrrm** [comment]
I think bitcoinj 0.15 is good to have but not sure how urgent it is. It is likely less costly to wait to merge since it's a separate library. Would be good to at least start testing it now and merge when we feel good about it.

@HarryMacfinned I think the main purpose of Bisq is to provide an on/off ramp to fiat that safeguards privacy and is censorship resistant. If we don't keep that focus the project will take a different path and new contributors won't have that focus. This basically makes it an altcoin trading platform. That seems like a bad idea since trading, as an activity, rather than as a mean to acquire bitcoin, is never going to be great on Bisq. The fact that it's used for alt coin trading only shows how our ethos speaks to people and the platform is good enough even for that use case. I think the project would die without the fiat focus.

My focus will most likely be on the new trade protocol going forward, but the scammer issue has to be handled first since fiat trading is at the core of what Bisq is.

**Comment by devinbileck** [comment]
> [bitcoinj 0.15] is likely less costly to wait to merge since it's a separate library.

There is a PR waiting to be merged into the main repo that touches a large number of files:
https://github.com/bisq-network/bisq/pull/2772

So like the API, the longer we wait the more maintenance required to keep in sync.

**Comment by sqrrm** [comment]
@devinbileck Thanks for the link, wasn't aware realize there is a proper PR to integrate it. Probably should get that done quite soon then.

**Comment by flix1** [comment]
@HarryMacfinned I agree with many of your concerns and thank you very much for stating them plainly.

> When I discuss about Bisq with some friends who trade, they always tell me that Bisq is already too complicated. And some of those guys are technical and should really imo be comfortable with Bisq.

This is something that I am specially worried about. Bisq is already too complicated, slow, etc. for _most_ potential users, which is already a small and technically-savvy population.

However I still believe that for the next 10 years the need for fiat-BTC exchange will dwarf the need for crypto-crypto and that if we manage to keep Bisq **uncensorable** it will have a huge market advantage in large parts of the world for a long time.

Also I think that you underestimate the risk of devoting resources to crypto-crypto market... only to see that market vanish completely when atomic swaps take off. The current XMR-BTC trade is actually just a handful of whales who could easily decide to go OTC.

**Comment by flix1** [comment]
Regarding the API I believe that it could go a long way not only to increase volumes, but to fix ux problems as it will open the possibility for other platforms to use Bisq as a backend while presenting users with a friendlier frontend.

**Comment by mpolavieja** [comment]
Closed as approved.
